libs/contrib/Data/Vect/Properties/Index.idr

```diff
@@ -0,0 +1,68 @@
+||| Properties of Data.Vect.index
+module Data.Vect.Properties.Index
+
+import Data.Vect.Properties.Tabulate
+
+import Data.Vect
+import Data.Vect.Elem
+import Data.Fin
+
+import Syntax.PreorderReasoning
+
+||| Recall an element by its position, as we may not have the element
+||| at runtime
+public export
+recallElem : {xs : Vect n a} -> x `Elem` xs -> a
+recallElem {xs = x :: _ }  Here         = x
+recallElem {xs = _ :: xs} (There later) = recallElem later
+
+||| Recalling by a position of `x` does yield `x`
+export
+recallElemSpec : (pos : x `Elem` xs) -> recallElem pos = x
+recallElemSpec  Here         = Refl
+recallElemSpec (There later) = recallElemSpec later
+
+||| map-fusion property for vectors up to function extensionality
+export
+mapFusionVect : (f : b -> c) -> (g : a -> b) -> (xs : Vect n a)
```

Comment:
I don't think we need the `Vect` suffix. You can always use a qualified
name if you think there is an ambiguity